cfgov/research_reports/models.py

```diff
@@ -0,0 +1,235 @@
+                          on_delete=models.CASCADE,
+                          related_name='report_sections_level_three')
+
+
+class ReportAuthor(models.Model):
+    name = models.CharField(max_length=50)
+    report = ParentalKey('ResearchReportPage',
+                         on_delete=models.CASCADE,
+                         related_name='report_authors')
+
+
+class ReportMetaclass(WagtailAdminModelFormMetaclass):
+    @classmethod
+    def child_form(cls):
+        return ReportForm
+
+
+class ReportForm(WagtailAdminPageForm, metaclass=ReportMetaclass):
+    pass
+
+
+class ResearchReportPage(CFGOVPage):
+    # Report Upload Fields
+    report_file = models.ForeignKey(
+        'wagtaildocs.Document',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+',
+    )
+    process_report = models.BooleanField(
+        default=False,
+        help_text=mark_safe(
+            'If this is checked, when you press "save", the system will '
+            'read in the report document and use its contents to overwrite '
+            'the fields in the "Report Content" tab.'
+            '<ul class="help">'
+            '    <li>&bull; If you uploaded a new report file for '
+            'processing, check this box.</li>'
+            '    <li>&bull; If you manually edited fields in the '
+            '"Report Content" tab, do not check this box</li>'
+            '</ul>'
+        )
+    )
+
+    # Report Upload Tabs
+    upload_panels = [
+        MultiFieldPanel([
+            FieldPanel('title'),
+        ], heading="Report Title"),
+        MultiFieldPanel([
+            DocumentChooserPanel('report_file'),
+            FieldPanel('process_report'),
+        ], heading='Report Document'),
+    ]
+
+    # Report Content Fields
+    base_form_class = ReportForm
+    report_type = models.CharField(max_length=100, blank=True)
+    header = models.CharField(max_length=200, blank=True)
+    subheader = models.TextField(blank=True)
+    pdf_location = models.CharField(max_length=150, blank=True)
+    footnotes = models.TextField(blank=True)
+
+    # Report content tab
+    content_panels = [
+        MultiFieldPanel([
+            FieldPanel('report_type'),
+            FieldPanel('header'),
+            FieldPanel('subheader'),
+            InlinePanel(
+                'report_authors',
+                label='Author Name',
+                min_num=0
+            ),
+            FieldPanel('pdf_location')
+        ], heading='Report Header'),
+        InlinePanel(
+            'report_sections',
+            label='Section',
+            min_num=0
+        ),
+        FieldPanel('footnotes')
+    ]
+
+    sidefoot_panels = CFGOVPage.sidefoot_panels
+
+    # Tab handler interface
+    edit_handler = TabbedInterface([
+        ObjectList(upload_panels, heading='Report Upload'),
+        ObjectList(content_panels, heading='Report Content'),
+        ObjectList(sidefoot_panels, heading='Sidebar'),
```

Comment:
```suggestion
        ObjectList(CFGOVPage.sidefoot_panels, heading='Sidebar'),
```

Doing this allows for deletion of the `sidefoot_panels` definition above.

Reply:
I meant to ask about this. @wpears Do we even want to allow sidefoot content? Seems like that's not part of the design, so could we leave out that tab entirely?

Reply:
Yeah I think we can nix the sidefoot

Reply:
Sweet. Done in b89e19f5ce6cc12.